src/executables/master.c

```diff
@@ -649,7 +651,7 @@ css_send_new_request_to_server (SOCKET server_fd, SOCKET client_fd, unsigned sho
  *   to the server.
  */
 static void
-css_send_to_existing_server (CSS_CONN_ENTRY * conn, unsigned short rid, CSS_SERVER_REQUEST request)
+css_send_to_existing_server (CSS_CONN_ENTRY * conn, unsigned short rid, CSS_SERVER_REQUEST request, SERVER_TYPE type)
```

Comment:
The description in the issue was wrong. The required server type will never be known when css_send_to_existing_server is called. It will be received along with the server_name by the next css_receive_data call.

Please revert this signature change and remove type. For now, css_return_entry_of_server may return any server.